pkg/sql/parser/testdata/create_table

```diff
@@ -590,6 +590,30 @@ CREATE TABLE a (b INT8 DEFAULT (1)) -- fully parenthesized
 CREATE TABLE a (b INT8 DEFAULT _) -- literals removed
 CREATE TABLE _ (_ INT8 DEFAULT 1) -- identifiers removed
 
+parse
+CREATE TABLE a (b INT8 ON UPDATE 1)
+----
+CREATE TABLE a (b INT8 ON UPDATE 1)
+CREATE TABLE a (b INT8 ON UPDATE (1)) -- fully parenthesized
+CREATE TABLE a (b INT8 ON UPDATE _) -- literals removed
+CREATE TABLE _ (_ INT8 ON UPDATE 1) -- identifiers removed
+
+error
+CREATE TABLE foo(b INT8 ON UPDATE 1 REFERENCES blah (blah) ON UPDATE SET NULL)
+----
+at or near ")": syntax error: ON UPDATE expression and ON UPDATE foreign key action both specified for column "b"
```

Comment:
we also need to do this for ALTER TABLE ADD FOREIGN KEY / SET ON UPDATE. but we can do that in a separate commit (please make an issue)

Reply:
Yep I'm planning on adding this in the next PR since you can't just do this at the parsing level. #69052